osmodel.mak

```diff
@@ -1,5 +1,15 @@
-# This Makefile snippet detects the OS and the architecture MODEL
-# Keep this file in sync between druntime, phobos, and dmd repositories!
+#   osmodel.mak
+#
+# Detects and sets the macros:
+#
+#   OS         = one of {osx,linux,freebsd,openbsd,netbsd,solaris}
+#   MODEL      = one of { 32, 64 }
+#   MODEL_FLAG = one of { -m32, -m64 }
+#
+# Note:
+#   Keep this file in sync between druntime, phobos, and dmd repositories!
```

Comment:
> I guess this needs accompanying PRs to dmd and phobos as per the comment.

FWIW Phobos and Druntime already depend on DMD and I see no logical reason to have redundant copies of this files...
I actually tried to push for this once:

https://github.com/dlang/phobos/pull/5228

But was stopped by people not wanting to have the DMD Makefile auto-checking out the tools repo.
However, we could go with the small scale solution and at least get rid of the osmodel files at Phobos, Druntime ...
Dlang.org was controversial as well due to automatic git clone